### ISSUE TITLE
Angular 2

### ISSUE THREAD

**Issue by imthehell** [danialfarid/ng-file-upload#1243]
Hello,

are you planning to update this great angular 1.x module to new angular 2?

Thank you guys

Nicolo

**Comment by Bartosh-s** [comment]
Very interested as well.

**Comment by mirikle** [comment]
+1

**Comment by LiboDing** [comment]
+1

**Comment by sweeneyrobb** [comment]
+1

**Comment by Jeff-Cortese** [comment]
+1

**Comment by KennyLisc** [comment]
+1

**Comment by harshaldhake** [comment]
+1

**Comment by root-io** [comment]
+1

**Comment by OliverKK** [comment]
+1

**Comment by AlexandreKilian** [comment]
Will probably be more of a complete rewrite than an update but +1

**Comment by arindambarman1** [comment]
Has anyone started on this? If not I would love to contribute or start one

**Comment by danialfarid** [comment]
No, Contributions are welcomed. Submit your PR here: https://github.com/danialfarid/ng2-file-upload

**Comment by arindambarman1** [comment]
Not being able to fork.

Regards,
Arindam Barman
arindambarman-twitter <https://twitter.com/arindambarman>
+919591374691

**Comment by danialfarid** [comment]
try again

**Comment by ncfx** [comment]
+1

**Comment by damianobarbati** [comment]
+999

**Comment by alvachien** [comment]
+1

**Comment by istiti** [comment]
+1

**Comment by danialfarid** [comment]
Hopefully before the end of the year.

**Comment by manishtuteja10** [comment]
@danialfarid - is ng-file-upload available now in angular 2 ?

**Comment by nihique** [comment]
+1

**Comment by teterovic** [comment]
+1

**Comment by teterovic** [comment]
+1

**Comment by teterovic** [comment]
@danialfarid Any updates ?

**Comment by Knoxvillekm** [comment]
+1

**Comment by danialfarid** [comment]
<b>A funding goal of $500 has been set for this issue.</b>
Donate via CodeFund.io to support this developer and solve this issue.
<a href='https://www.codefund.io/view.php?id=13'><img src='http://www.codefund.io/img/percent/0-percent-funded.png'></a>
<a href='https://www.codefund.io/view.php?id=13'><b>Make a Donation!</b></a>
<i>Why donate? Learn more at <a href='https://www.codefund.io/support-open-source.php'>www.codefund.io/support-open-source.php</a></i>

**Comment by moustafa20** [comment]
+1

**Comment by wjbLabs** [comment]
Codefund site is dead.

**Comment by georgediaz88** [comment]
@danialfarid,  first `ng-file-upload` is an awesome library that's helped us upload files with ease and it's super robust. So, thanks for your contributions! 

I'm curious like everyone on this thread if there's any update on when this library will support Angular >= 2 so I could use it for modern apps. Further, I noticed an [ng2](https://github.com/danialfarid/ng-file-upload/tree/ng2) branch. If I need to wait some, do you consider that branch production ready / about done? That way I can use that in the interim.

Please let me know and thanks again!

**Comment by danialfarid** [comment]
The branch should be stable enough and working but I would need more time to updating the docs, providing demo, automating builds, etc which could take a while since I am preoccupied with other works. Any help or contribution would be really appreciated.

**Comment by frostme** [comment]
@danialfarid are you saying ng2 branch is working now for angular 2+ environments? did you have a chance at all to work on the docs?

**Comment by danialfarid** [comment]
@frostme That is correct. Not yet but most syntax is about the same and there is a demo folder to see the different samples in that branch.
